src/components/Clock.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+interface Props {
+  name: string;
+}
+
+type State = {
+  today: Date;
+};
+
+function correctTime(date: Date): string {
+  return date.toUTCString().slice(-12, -4);
+}
+
+export class Clock extends React.Component<Props, State> {
+  state: Readonly<State> = {
+    today: new Date(),
+  };
+
+  timer = 0;
+
+  componentDidMount() {
+    this.timer = window.setInterval(() => {
+      this.setState({ today: new Date() });
+      window.console.info(correctTime(this.state.today));
```

Comment:
don't use state here, it can be outdated
save `new Date` result to constant and use it in both setState and console.info